taxonomy.html

```diff
@@ -0,0 +1,478 @@
+          to all roles in the taxonomy.
+        </p>
+        <pre class="example highlight">
+var library = {
+  roles: {
+    '*': {
+      states: [
+        'highlighable', 'highlighted'
+      ],
+      relations: [
+        'hihglight-comment-for',
+        'hihglight-commented-by'
+      ]
+    }
+  },
+  states: {
+    highlighable: { },
+    highlighted: {
+      description: 'user choice'
+    }
+  },
+  relations: {
+    'hihglight-comment-for': { },
+    'hihglight-commented-by': {
+      description: 'commented by'
+    }
+  }
+}
+</pre>
+      </section>
+    </section>
+    <section>
+      <h2>
+        TaxoLibrary dictionary
+      </h2>
+      <p>
+        A collection of taxonomies extensions to the document's taxonomies.
+      </p>
+      <pre class="idl">
+dictionary TaxoLibrary {
+  DOMString id;
+  object roles;
+  object states;
+  object attributes;
+  object actions;
+  object relations;
+};
+</pre>
+      <section>
+        <h3>
+          <code>id</code> attribute
+        </h3>
+        <p>
+          The <code>id</code> attribute represents the library id, used for
+          identification proposes.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <code>roles</code> attribute
+        </h3>
+        <p>
+          The <code>roles</code> attribute represents a collection of
+          <a>roles</a> added by the library into browser's taxonomy. A role is
+          described by a dictionary repeating of <code>RoleTaxon</code>
+          interface.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <code>states</code> attribute
+        </h3>
+        <p>
+          The <code>states</code> attribute represents a collection of
+          <a>states</a> added by the library into browser's taxonomy. A state
+          is described by a dictionary repeating properties of
+          <code>Taxon</code> interface.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <code>attributes</code> attribute
+        </h3>
+        <p>
+          The <code>attributes</code> attribute represents a collection of
+          <a>attributes</a> added by the library into browser's taxonomy. An
+          attribute is described by a dictionary repeating properties of
+          <code>Taxon</code> interface.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <code>relations</code> attribute
+        </h3>
+        <p>
+          The <code>relations</code> attribute represents a collection of
+          <a>relations</a> added by the library into browser's taxonomy. A
+          relation is described by a dictionary repeating properties of
+          <code>Taxon</code> interface.
+        </p>
+      </section>
+    </section>
+    <section>
+      <h2>
+        <code>Taxon</code> interface
+      </h2>
+      <p>
+        An object containing a name and description of a taxon.
+      </p>
+      <pre class="idl">
+[NoInterfaceObject]
```

Comment:
Avoid using [NoInterfaceObject], unless absolutely necessary... othewise it makes it hard for devs to polyfill functionality.